nulltime.go

```diff
@@ -1,50 +0,0 @@
-// Go MySQL Driver - A MySQL-Driver for Go's database/sql package
-//
-// Copyright 2013 The Go-MySQL-Driver Authors. All rights reserved.
-//
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this file,
-// You can obtain one at http://mozilla.org/MPL/2.0/.
-
-package mysql
-
-import (
-	"database/sql/driver"
-	"fmt"
-	"time"
-)
-
-// Scan implements the Scanner interface.
-// The value type must be time.Time or string / []byte (formatted time-string),
-// otherwise Scan fails.
-func (nt *NullTime) Scan(value interface{}) (err error) {
-	if value == nil {
-		nt.Time, nt.Valid = time.Time{}, false
-		return
-	}
-
-	switch v := value.(type) {
-	case time.Time:
-		nt.Time, nt.Valid = v, true
-		return
-	case []byte:
```

Comment:
`parseDateTime` for `[]byte` and `string` are still required, even with sql.NullTime in Go 1.3+.

Reply:
I think it is misuse of mysqlNullTime, users should add the `parseTime=true` option.
because `parseDateTime` does not consider timezones.
`TestLegacyNullTime` and `TestNullTime` are working examples.

but, you know, it may break existing codes. I'm thinking about it now... 🤔

Reply:
We should at least not push it right before a release. Let's reconsider this later.

Reply:
Agree